hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/server/TopicMetadataLoadingJob.java

```diff
@@ -0,0 +1,53 @@
+package pl.allegro.tech.hermes.frontend.server;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import pl.allegro.tech.hermes.common.config.ConfigFactory;
+import pl.allegro.tech.hermes.common.config.Configs;
+
+import javax.inject.Inject;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+public class TopicMetadataLoadingJob implements Runnable {
+
+    private final TopicMetadataLoadingRunner topicMetadataLoadingRunner;
+    private final ScheduledExecutorService executorService;
+    private final int intervalSeconds;
+
+    private ScheduledFuture job;
+
+    @Inject
+    public TopicMetadataLoadingJob(TopicMetadataLoadingRunner topicMetadataLoadingRunner,
+                                   ConfigFactory config) {
+        this(topicMetadataLoadingRunner, config.getIntProperty(Configs.FRONTEND_TOPIC_METADATA_REFRESH_JOB_INTERVAL_SECONDS));
+    }
+
+    TopicMetadataLoadingJob(TopicMetadataLoadingRunner topicMetadataLoadingRunner, int intervalSeconds) {
+        this.topicMetadataLoadingRunner = topicMetadataLoadingRunner;
+        this.intervalSeconds = intervalSeconds;
+
+        ThreadFactory threadFactory = new ThreadFactoryBuilder()
+                .setNameFormat("TopicMetadataLoadingJob-%d").build();
+        this.executorService = Executors.newSingleThreadScheduledExecutor(threadFactory);
+    }
+
+    @Override
+    public void run() {
+        topicMetadataLoadingRunner.refreshMetadata();
+    }
+
+    public void start() {
+        job = executorService.scheduleAtFixedRate(this, intervalSeconds, intervalSeconds, TimeUnit.SECONDS);
```

Comment:
I think if `initialDelay` (second argument) would be equal to 0 instead of `intervalSeconds` then we wouldn't need `TopicMetadataLoadingStartupHook`.

Reply:
`TopicMetadataLoadingStartupHook` plays a specific role when booting frontends, it blocks readiness health checks from becoming green, i.e. it is required to load all topics metadata before we start accepting incoming traffic. Here it would be just async, so it's a different case.

Reply:
You can disable refreshing job by setting `FRONTEND_TOPIC_METADATA_REFRESH_JOB_ENABLED` to `false`, but metadata should always be loaded on startup

Reply:
actually there's `frontend.startup.topic.metadata.loading.enabled` flag, but anyway... :) IMO it's different purpose.